### DISCUSSION TITLE
Trying to add new feature - System.CommandLine is confuzzeling me

### DISCUSSION THREAD

**Discussion by RachelAmbler** [grate-devs/grate#238] [Q&A] [answered]
I _thought_ I'd got everything correct - but for some reason my new command line arguments are not being parsed:

In `MigrateCommand` I have

```csharp
 //DEPENDENCY OPTIONS
private static Option<bool> AnalyzeScriptsForDependencies() =>
    new(
        new[] { "--dependencies", "--de" },
        //getDefaultValue: () => false,
        description:"AnalyzeScriptsForDependencies - This instructs grate that each script may contain dependency specifications."
    );
```

and above that I've added the option to `MigrateCommand` with

```csharp
public MigrateCommand(GrateMigrator mi) : base("Migrates the database")
    {
        Add(Database());
        Add(ConnectionString());
        Add(SqlFilesDirectory());
        ...
        Add(AnalyzeScriptsForDependencies());
       

```

In `GrateConfiguration` I have

```csharp
/// <summary>
/// If specified, inform grate to look for dependencies in scripts to define the order they should be executed in
/// </summary>
public bool AnalyzeScriptsForDependencies { get; init; } = false;
```

However my tests continue to fail in `Basic_CommandLineParsing`:
```csharp
[TestCase("", false)] // default
[TestCase("--dependencies=true", true)]
[TestCase("--dependencies true", true)]
[TestCase("--dependencies", true)]
public async Task TestAnalyzeScriptsForDependencies(string args, bool expected)
{
    var cfg = await ParseGrateConfiguration(args);
    cfg?.AnalyzeScriptsForDependencies.Should().Be(expected);
}
```

The first test succeeds (default to `false`) but all the others fail (still set to `false`)

I've actually got three options in total I've added, but if I can get this first one to work, the other two should be a doddle.

What on earth am I doing wrong here? Have I missed something out?

**Comment by RachelAmbler** [comment, marked as answer]
Sussed it:
```csharp
//DEPENDENCY OPTIONS
private static Option<bool> AnalyzeScriptsForDependencies() =>
    new(
        new[] { "--AnalyzeScriptsForDependencies", "--dependencies", "--de" },
        description:"AnalyzeScriptsForDependencies - This instructs grate that each script may contain dependency specifications."
    );
```
The first argument (or one of the arguments at least) apparently has to match the option name.

**Reply by erikbra** [reply]
Hi, @RachelAmbler ! Yes, those System.CommandLine things are a bit difficult from time to time. I always forget how they work between each time I use them.

The option is very interesting, though. Have you got anywhere with it? What is your approach for analysing dependencies?

**Reply by RachelAmbler** [reply]
> Hi, @RachelAmbler ! Yes, those System.CommandLine things are a bit difficult from time to time. I always forget how they work between each time I use them.
> 
> The option is very interesting, though. Have you got anywhere with it? What is your approach for analysing dependencies?

Think you found my update!

Yeah - once I discovered "The secret source" it made sense. Any objection if I create a page in the documentation to describe this for anyone else?

**Reply by erikbra** [reply]
No, by all means, please do! Any help in getting started with extending, improving, developing grate is greatly appreaciated!
